h2/settings.py

```diff
@@ -254,12 +254,18 @@ def __len__(self):
         return len(self._settings)
 
 
-    def __eq__(self,other):
-        return self._settings==other._settings
-
-    def __ne__(self,other):
-        return not self._settings==other._settings
-
+    def __eq__(self, other):
+        if isinstance(other, Settings):
+            return self._settings == other._settings
+        else:
+            return NotImplemented
```

Comment:
If `other` isn't a `Settings` object we can pretty easily say we're not equal. Should return `False` in this case.